internal/middleware/region.go

```diff
@@ -16,12 +14,11 @@ const (
 	requestRegionKey key = iota + 1
 )
 
-func AddRegionToContext(defaultRegion string) mux.MiddlewareFunc {
+func AddRegionToContext(defaultRegion string) MiddlewareFunc {
```

Comment:
Somehow this middleware is called twice before the request reaches dedicated endpoint.

Reply:
Fixed. The problem was that the middleware set for the parent router was also called for the child router. I moved the middleware definition to the child router since it is not used by any endpoints from the parent router.